### ISSUE TITLE
Reason for using LARGE_NUMBER

### ISSUE THREAD

**Issue by rohan1561** [AndrewAtanov/simclr-pytorch#13]
Hello, thanks for the great work! I was wondering what the reason must be behind using the self.LARGE_NUMBER. I understand that it serves to suppress the logits due to self multiplication but is it really necessary given that the labels are negative for them anyways?
Thanks!

**Comment by AndrewAtanov** [comment]
Hello @rohan1561, it's been a while and I might be missing something, but the idea is to set the corresponding probabilities to 0 so they effectively do not appear in the logsumexp [here](https://github.com/AndrewAtanov/simclr-pytorch/blob/d147c6bea1787e0d68dd334327c93e69b56f601e/models/losses.py#L76)
